### ISSUE TITLE
Issue with TEEC_AllocateSharedMemory for Buffer Size Greater Than 512KB on QEMU Platform

### ISSUE THREAD

**Issue by J-StrawHat** [OP-TEE/optee_os#6225]
I've encountered an issue while trying to allocate a data buffer greater than 512KB (e.g., 1MB) using the `TEEC_AllocateSharedMemory()` on the **qemu** platform. When attempting this allocation, the function returns the error code `0xffff000c`. However, when I allocate a buffer of 512KB or less, the operation succeeds without any issues.

main.c in CA :
```c
TEEC_SharedMemory sharedBuf = {0};
sharedBuf.flags = TEEC_MEM_INPUT;
sharedBuf.size = 1048576;
TEEC_Result res;
res = TEEC_AllocateSharedMemory(&ta.ctx, &sharedBuf);
if (res != TEEC_SUCCESS)
    errx(1, "\nTEEC_AllocateSharedMemory failed with code 0x%x\n", res);
```

user_ta_header_defines.h :
```c
#define TA_STACK_SIZE (2 * 1024 * 1024)
#define TA_DATA_SIZE (3 * 1024 * 1024)
```

optee_os/core/arch/arm/plat-vexpress/conf.mk : 
```
CFG_TZDRAM_SIZE  ?= 0x01100000
CFG_SHMEM_SIZE  ?= 0x02000000
```

Could you please provide guidance on how I can allocate larger memory sizes? I would appreciate any insight or suggestions on how to handle this situation.

**Comment by J-StrawHat** [comment]
Any suggestions for this issue?

**Comment by jenswi-linaro** [comment]
I assume that you've modified the configuration in the standard QEMU setup. You could try using an unmodified configuration to see if that works as expected.

**Comment by J-StrawHat** [comment]
I'm so sorry I didn't see your suggestions until now.

Indeed, I changed the `QEMU_CPU` in qemu_v8.mk to `cortex-a72` instead of `cortex-a57`, which makes me encountered the problems mentioned above.

When I used an unmodified configuration in the standard QEMU setup and modifed the `user_ta_header_defines.h`, I can allocate larger memory sizes (8MB). 

Thank you very much. This question has been bothering me for a long time.
